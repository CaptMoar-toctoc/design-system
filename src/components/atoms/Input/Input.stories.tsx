import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Input from './Input';

export default {
	title: 'TocToc-DS/Atoms',
	component: Input,
	argTypes: {
		//checked:{control: 'boolean'}
		size: {
			control: {
				type: 'inline-radio',
				options: ['small', 'medium'],
			},
		},
	},
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const InputTT = Template.bind({});


InputTT.args = {
	label: "Ingresa un dato",
	disabled: false,
	type: "text",
	defaultValue: "",
	error: false,
	fullWidth: false,
	helperText: "Texto de ayuda para este campo.",
	size: "medium",
	name:"nombreInput",
	onChange: () => { console.log(`onChange input`) },
	customStyleObject: {},
	value: "",
};
InputTT.decorators = [
	(Story) => (
		<div style={{ maxWidth: '30%' }}>
			<Story />
		</div>
	),
];
