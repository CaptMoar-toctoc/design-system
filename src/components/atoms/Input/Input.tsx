import React, { FC } from 'react';
import { iProps } from './iProps';
import { ThemeProvider, TextField as InputTT } from '@mui/material';

import theme from '../../theme/theme';

const Input: FC<iProps> = ({
	onChange,
	label,
	type,
	disabled,
	customStyleObject,
	defaultValue,
	error,
	fullWidth,
	helperText,
	size,
	value,
	name
}) => {
	return (
		<ThemeProvider theme={theme}>

			<InputTT
				label={label}
				value={value}
				type={type}
				sx={customStyleObject}
				disabled={disabled}
				defaultValue={defaultValue}
				error={error}
				fullWidth={fullWidth}
				helperText={helperText}
				size={size}
				name={name}
				onChange={onChange}
			/>

		</ThemeProvider>
	);
}

export default Input;