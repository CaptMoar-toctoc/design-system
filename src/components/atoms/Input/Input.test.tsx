import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';


import Input from "./Input";

describe('Input Testing Render', () => {
	const props = {
		label: 'Holaaaa',
		type: 'number',
		error: false,
		fullWidth: true,
		helperText: 'texto ayuda input',
		size: 'medium',
		value: 'valor input',
		name: 'nameInput',
		onChange: () => {  },

	}

	test("Render SnapShot", () => {
		expect(render(<Input {...props} />)).toMatchSnapshot();
	})

	test("Validar que sea input text", () => {
		const { container } = render(<Input {...props} />);
		const foo = container.querySelector(`[type="${props.type}"]`) as HTMLInputElement;
		expect(foo?.type).toBe(props.type);
	})
})

