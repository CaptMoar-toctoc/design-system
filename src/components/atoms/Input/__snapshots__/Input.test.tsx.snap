// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Testing Render Render SnapShot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-1h2568w-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Holaaaa
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1qoyjwj-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-describedby=":r0:-helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-owdxj3-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            name="nameInput"
            type="number"
            value="valor input"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Holaaaa
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-idv3ut-MuiFormHelperText-root"
          id=":r0:-helper-text"
        >
          texto ayuda input
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-1h2568w-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":r0:"
        id=":r0:-label"
      >
        Holaaaa
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1qoyjwj-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-describedby=":r0:-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-owdxj3-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          name="nameInput"
          type="number"
          value="valor input"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1ftyaf0"
          >
            <span>
              Holaaaa
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-idv3ut-MuiFormHelperText-root"
        id=":r0:-helper-text"
      >
        texto ayuda input
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
